internals/workloads/sectionextension.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2025 Canonical Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a Copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workloads
+
+import (
+	"fmt"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/canonical/pebble/internals/osutil"
+	"github.com/canonical/pebble/internals/plan"
+)
+
+var _ plan.SectionExtension = (*SectionExtension)(nil)
+
+type SectionExtension struct{}
+
+func (ext *SectionExtension) CombineSections(sections ...plan.Section) (plan.Section, error) {
+	ws := &WorkloadsSection{}
+	for _, section := range sections {
+		// The following will panic if any of the supplied section is not a WorkloadsSection
```

Comment:
typo: ```... any if the supplied sections are ... ```

I wonder how useful this comment is ? This will only occur in the case of some coding error, and then the behavior will be obvious enough, maybe ?

Reply:
About the internal error, I literally copied that from you, but Ben rightfully pointed out that it introduced a bit of noise that gave little to no value, so I just let the type assertion panic because the developer (us) is meant to know when to use it properly

Reply:
Sorry my comment was unclear. I do not think the internal error is worth it, but I was just asking if the comment about an type assertion causing a panic (as apposed many other situations in code that can also cause panics) is worth explicitly documenting.

Reply:
I added it as a suggestion from (I think) Ben